telemetry/main.go

```diff
@@ -0,0 +1,110 @@
+package telemetry
+
+import (
+	"context"
+	"log/slog"
+	"net/http"
+	"os"
+
+	"go.opentelemetry.io/contrib/detectors/aws/ecs"
+	"go.opentelemetry.io/contrib/propagators/aws/xray"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/sdk/trace"
+	oteltrace "go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+)
+
+type contextKey string
```

Comment:
most of the time people will do `type contextKey struct{}` to avoid allocating a string, and here makes sense since there is only one key

Reply:
Done, I've also renamed to `loggerContextKey` for a bit of internal clarity (unless I've missed the point of the struct)